javav2/usecases/topics_and_queues/src/main/java/com/example/sns/SNSWorkflow.java

```diff
@@ -0,0 +1,669 @@
+        atts.add(QueueAttributeName.QUEUE_ARN);
+
+        GetQueueAttributesRequest attributesRequest= GetQueueAttributesRequest.builder()
+            .queueUrl(queueUrl)
+            .attributeNames(atts)
+            .build();
+
+        GetQueueAttributesResponse response = sqsClient.getQueueAttributes(attributesRequest);
+        Map<String,String> queueAtts = response.attributesAsStrings();
+        for (Map.Entry<String,String> queueAtt : queueAtts.entrySet())
+            return queueAtt.getValue();
+
+        return "";
+    }
+
+    public static String createQueue(SqsClient sqsClient,String queueName, Boolean selectFIFO ) {
+        try {
+            System.out.println("\nCreate Queue");
+            if (selectFIFO) {
+                Map<QueueAttributeName, String> attrs = new HashMap<>();
+                attrs.put(QueueAttributeName.FIFO_QUEUE, "true");
+                CreateQueueRequest createQueueRequest = CreateQueueRequest.builder()
+                    .queueName(queueName)
+                    .attributes(attrs)
+                    .build();
+
+                sqsClient.createQueue(createQueueRequest);
+                System.out.println("\nGet queue url");
+                GetQueueUrlResponse getQueueUrlResponse = sqsClient.getQueueUrl(GetQueueUrlRequest.builder().queueName(queueName).build());
+                return getQueueUrlResponse.queueUrl();
+            } else {
+                CreateQueueRequest createQueueRequest = CreateQueueRequest.builder()
+                        .queueName(queueName)
+                        .build();
+
+                sqsClient.createQueue(createQueueRequest);
+                System.out.println("\nGet queue url");
+                GetQueueUrlResponse getQueueUrlResponse = sqsClient.getQueueUrl(GetQueueUrlRequest.builder().queueName(queueName).build());
+                return getQueueUrlResponse.queueUrl();
+           }
+
+        } catch (SqsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+        return "";
+    }
+
+    public static String createSNSTopic(SnsClient snsClient, String topicName ) {
+        CreateTopicResponse result;
+        try {
+            CreateTopicRequest request = CreateTopicRequest.builder()
+                .name(topicName)
+                .build();
+
+            result = snsClient.createTopic(request);
+            return result.topicArn();
+
+        } catch (SnsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+        return "";
+    }
+
+    public static String createFIFO(SnsClient snsClient, String topicName) {
+        try {
+            // Create a FIFO topic by using the SNS service client.
+            Map<String, String> topicAttributes = new HashMap<>();
+            topicAttributes.put("FifoTopic", "true");
+            topicAttributes.put("ContentBasedDeduplication", "false");
```

Comment:
You always have ContentBasedDeduplication false, but the user was asked if they want content-based deduplication.,